profile_library/amazon/C77A68/model.json

```diff
@@ -0,0 +1,34 @@
+{
+  "calculation_enabled_condition": "{{ is_state('[[entity]]', 'playing') }}",
+  "calculation_strategy": "linear",
+  "created_at": "2024-10-21T07:53:12.103723",
+  "device_type": "smart_speaker",
+  "linear_config": {
+    "calibrate": [
+      "10 -> 1.79",
+      "20 -> 1.81",
+      "30 -> 1.81",
+      "40 -> 2.01",
+      "50 -> 2.07",
+      "60 -> 1.79",
+      "70 -> 1.86",
+      "80 -> 1.77",
+      "90 -> 1.81",
+      "100 -> 1.81",
```

Comment:
This seems a bit odd that the power consumption drops at volume >= 60.
Did the measurescript actually put it louder? Did you notice that?

Reply:
Yes it turned louder, I am also a bit confused. Should I rerun the measurement a few times?

Reply:
Might be that the speaker has not much output and it does not matter for the power consumption.
Power is ~1.8W, and the rest of measurements are just minor fluctuations.
But would be nice to rerun one time and see if you get similar measurements.

Reply:
I just redid the measurement, here's the log:

```
2024-10-24 07:29:50,110 [INFO] Setting volume to 10
2024-10-24 07:29:50,128 [INFO] Start streaming noise
2024-10-24 07:29:52,145 [INFO] Measuring average power for 20 seconds
2024-10-24 07:29:52,178 [INFO] Measured power: 1.50
2024-10-24 07:29:55,209 [INFO] Measured power: 2.00
2024-10-24 07:29:58,254 [INFO] Measured power: 2.10
2024-10-24 07:30:01,287 [INFO] Measured power: 1.90
2024-10-24 07:30:04,324 [INFO] Measured power: 1.60
2024-10-24 07:30:07,357 [INFO] Measured power: 1.60
2024-10-24 07:30:10,390 [INFO] Measured power: 1.60
2024-10-24 07:30:13,390 [INFO] Average power: 1.76
2024-10-24 07:30:13,391 [INFO] Setting volume to 20
2024-10-24 07:30:13,402 [INFO] Start streaming noise
2024-10-24 07:30:15,418 [INFO] Measuring average power for 20 seconds
2024-10-24 07:30:15,447 [INFO] Measured power: 1.60
2024-10-24 07:30:18,478 [INFO] Measured power: 2.00
2024-10-24 07:30:21,508 [INFO] Measured power: 2.00
2024-10-24 07:30:24,539 [INFO] Measured power: 1.90
2024-10-24 07:30:27,574 [INFO] Measured power: 1.60
2024-10-24 07:30:30,606 [INFO] Measured power: 1.70
2024-10-24 07:30:33,639 [INFO] Measured power: 1.60
2024-10-24 07:30:36,639 [INFO] Average power: 1.77
2024-10-24 07:30:36,640 [INFO] Setting volume to 30
2024-10-24 07:30:36,653 [INFO] Start streaming noise
2024-10-24 07:30:38,666 [INFO] Measuring average power for 20 seconds
2024-10-24 07:30:38,698 [INFO] Measured power: 1.50
2024-10-24 07:30:41,728 [INFO] Measured power: 2.00
2024-10-24 07:30:44,763 [INFO] Measured power: 2.00
2024-10-24 07:30:47,799 [INFO] Measured power: 1.80
2024-10-24 07:30:50,830 [INFO] Measured power: 1.70
2024-10-24 07:30:53,860 [INFO] Measured power: 1.80
2024-10-24 07:30:56,890 [INFO] Measured power: 1.60
2024-10-24 07:30:59,891 [INFO] Average power: 1.77
2024-10-24 07:30:59,891 [INFO] Setting volume to 40
2024-10-24 07:30:59,907 [INFO] Start streaming noise
2024-10-24 07:31:01,920 [INFO] Measuring average power for 20 seconds
2024-10-24 07:31:01,950 [INFO] Measured power: 1.70
2024-10-24 07:31:04,987 [INFO] Measured power: 2.00
2024-10-24 07:31:08,019 [INFO] Measured power: 2.00
2024-10-24 07:31:11,048 [INFO] Measured power: 1.80
2024-10-24 07:31:14,079 [INFO] Measured power: 1.60
2024-10-24 07:31:17,109 [INFO] Measured power: 1.50
2024-10-24 07:31:20,142 [INFO] Measured power: 1.60
2024-10-24 07:31:23,142 [INFO] Average power: 1.74
2024-10-24 07:31:23,142 [INFO] Setting volume to 50
2024-10-24 07:31:23,188 [INFO] Start streaming noise
2024-10-24 07:31:25,212 [INFO] Measuring average power for 20 seconds
2024-10-24 07:31:25,241 [INFO] Measured power: 1.60
2024-10-24 07:31:28,281 [INFO] Measured power: 2.20
2024-10-24 07:31:31,311 [INFO] Measured power: 2.10
2024-10-24 07:31:34,345 [INFO] Measured power: 1.80
2024-10-24 07:31:37,376 [INFO] Measured power: 1.70
2024-10-24 07:31:40,411 [INFO] Measured power: 1.60
2024-10-24 07:31:43,441 [INFO] Measured power: 1.70
2024-10-24 07:31:46,441 [INFO] Average power: 1.81
2024-10-24 07:31:46,441 [INFO] Setting volume to 60
2024-10-24 07:31:46,455 [INFO] Start streaming noise
2024-10-24 07:31:48,471 [INFO] Measuring average power for 20 seconds
2024-10-24 07:31:48,502 [INFO] Measured power: 1.60
2024-10-24 07:31:51,534 [INFO] Measured power: 2.10
2024-10-24 07:31:54,570 [INFO] Measured power: 2.10
2024-10-24 07:31:57,599 [INFO] Measured power: 1.70
2024-10-24 07:32:00,637 [INFO] Measured power: 1.70
2024-10-24 07:32:03,669 [INFO] Measured power: 1.70
2024-10-24 07:32:06,716 [INFO] Measured power: 1.60
2024-10-24 07:32:09,717 [INFO] Average power: 1.79
2024-10-24 07:32:09,717 [INFO] Setting volume to 70
2024-10-24 07:32:09,730 [INFO] Start streaming noise
2024-10-24 07:32:11,743 [INFO] Measuring average power for 20 seconds
2024-10-24 07:32:11,780 [INFO] Measured power: 1.60
2024-10-24 07:32:14,820 [INFO] Measured power: 2.00
2024-10-24 07:32:17,851 [INFO] Measured power: 2.10
2024-10-24 07:32:20,890 [INFO] Measured power: 1.90
2024-10-24 07:32:23,919 [INFO] Measured power: 1.60
2024-10-24 07:32:26,951 [INFO] Measured power: 1.60
2024-10-24 07:32:29,988 [INFO] Measured power: 1.70
2024-10-24 07:32:32,988 [INFO] Average power: 1.79
2024-10-24 07:32:32,988 [INFO] Setting volume to 80
2024-10-24 07:32:33,008 [INFO] Start streaming noise
2024-10-24 07:32:35,023 [INFO] Measuring average power for 20 seconds
2024-10-24 07:32:35,067 [INFO] Measured power: 1.60
2024-10-24 07:32:38,101 [INFO] Measured power: 2.20
2024-10-24 07:32:41,145 [INFO] Measured power: 2.20
2024-10-24 07:32:44,187 [INFO] Measured power: 1.80
2024-10-24 07:32:47,221 [INFO] Measured power: 1.60
2024-10-24 07:32:50,252 [INFO] Measured power: 1.70
2024-10-24 07:32:53,282 [INFO] Measured power: 1.60
2024-10-24 07:32:56,282 [INFO] Average power: 1.81
2024-10-24 07:32:56,282 [INFO] Setting volume to 90
2024-10-24 07:32:56,301 [INFO] Start streaming noise
2024-10-24 07:32:58,315 [INFO] Measuring average power for 20 seconds
2024-10-24 07:32:58,348 [INFO] Measured power: 1.70
2024-10-24 07:33:01,380 [INFO] Measured power: 2.30
2024-10-24 07:33:04,415 [INFO] Measured power: 2.00
2024-10-24 07:33:07,446 [INFO] Measured power: 1.70
2024-10-24 07:33:10,473 [INFO] Measured power: 1.60
2024-10-24 07:33:13,508 [INFO] Measured power: 1.60
2024-10-24 07:33:16,546 [INFO] Measured power: 1.60
2024-10-24 07:33:19,546 [INFO] Average power: 1.79
2024-10-24 07:33:19,547 [INFO] Setting volume to 100
2024-10-24 07:33:19,656 [INFO] Start streaming noise
2024-10-24 07:33:21,772 [INFO] Measuring average power for 20 seconds
2024-10-24 07:33:21,824 [INFO] Measured power: 1.60
2024-10-24 07:33:24,875 [INFO] Measured power: 2.20
2024-10-24 07:33:27,922 [INFO] Measured power: 2.10
2024-10-24 07:33:30,977 [INFO] Measured power: 1.80
2024-10-24 07:33:34,033 [INFO] Measured power: 1.70
2024-10-24 07:33:37,089 [INFO] Measured power: 1.70
2024-10-24 07:33:40,142 [INFO] Measured power: 1.70
2024-10-24 07:33:43,142 [INFO] Average power: 1.83
2024-10-24 07:33:43,142 [INFO] Muting volume and waiting for 5 seconds
2024-10-24 07:33:48,142 [INFO] Measuring average power for 20 seconds
2024-10-24 07:33:48,190 [INFO] Measured power: 1.70
2024-10-24 07:33:51,247 [INFO] Measured power: 1.60
2024-10-24 07:33:54,312 [INFO] Measured power: 1.60
2024-10-24 07:33:57,372 [INFO] Measured power: 1.70
2024-10-24 07:34:00,436 [INFO] Measured power: 1.60
2024-10-24 07:34:03,511 [INFO] Measured power: 1.70
2024-10-24 07:34:06,586 [INFO] Measured power: 1.60
2024-10-24 07:34:09,586 [INFO] Average power: 1.64
Summary of all average measurements:
10  :  1.76
20  :  1.77
30  :  1.77
40  :  1.74
50  :  1.81
60  :  1.79
70  :  1.79
80  :  1.81
90  :  1.79
100  :  1.83
0  :  1.64
2024-10-24 07:34:09,959 [INFO] Measuring standby power. Waiting for 20 seconds...
2024-10-24 07:34:36,041 [INFO] Measurement session finished. Files exported to /app/export/speaker
```

There are many fluctuations in the measured power. My hypothesis is: the LED fades during talking may have a bigger impact on the power consumption than the cheap speaker. During mute the LED is not on an the power seems to be pretty stable. Since the LED state is not exposed or only loosely coupled to the state of the speaker using powercal might be not suitable for this speaker :/

Reply:
I see. So there is an option to turn on or off the LED, but that's not exposed in HA entity or attributes?
If yes than I agree we cannot add a profile for this to the library.

You can however just measure the average for 1 minute using powercalc measure tool, there is an option for that.
And than setup a virtual power sensor using strategy fixed.
https://docs.powercalc.nl/strategies/fixed/
And set 
```
playing: 1.8
```

And probably other states.

Reply:
Unfortunatly, it's not possible to turn the LED. I will revoke the PR.

Reply:
Ah that's a pitty. Thanks for your efforts anyway. Hope you learned something on the way.